wp/lib/bap_wp/src/environment.ml

```diff
@@ -125,6 +130,41 @@ let wp_rec_call :
   (t -> Constr.t -> start:Graphs.Ir.Node.t -> Graphs.Ir.t -> t) ref =
   ref (fun _ _ ~start:_ _ -> assert false)
 
+let trivial_pre (env : t) : Constr.t =
+  let ctx = get_context env in
+  Z3.Boolean.mk_true ctx
+  |> Constr.mk_goal "true"
+  |> Constr.mk_constr
+
+let post_dominator (env : t) (node : Graphs.Ir.Node.t) (graph : Graphs.Ir.t)
```

Comment:
This function is weirdly named: is it getting the precondition of the loop exit? If so, consider naming it `get_loop_exit_pre` or something similar.

Reply:
Also maybe useful to give a short summary of what's going on, for future uses sake...